### PR TITLE
Java 8 app migrated to Java 11

### PR DESCRIPTION
**Pull request by averikitsch**
Embedded jetty server + servlet
run locally by mvn exec:java

Are these the dependencies we want to show?
How should we package this for deployment?

### CODE REVIEW

**Comment by lesv** [comment]
That said, I can see why you don't want to in this instance.  I just hate making things more difficult on users. I could argue both ways, To me, the real issue is do we want jetty running inside maven and how users will debug using an IDE connecting up to the app.  Ray will give you good advice.

**Comment by averikitsch** [comment]
@lesv Thanks for the comments! Ray and I have decided to go in a different direction with this.

**Comment by averikitsch** [comment]
@lesv Do you mind reviewing?

File: appengine-java11/README.md [hunk 1]
```diff
@@ -46,11 +46,29 @@ To switch to an Open JDK 11 in a Cloud shell session, you can use:
...
 jar, the Main Class will load a war file, passed as an argument, as the
 context root of the web application listening to port 8080.
 Some samples create a `<sample-name>.war` which is used as an argument in the
-App Engine app.yaml entrypoint field.
+App Engine `app.yaml` entrypoint field.
+
+### App Engine Staging Directory
+
+The App Engine Plugin will stage all the files to upload into App Engine
+runtime in `${build.directory}/appengine-staging`. When deploying an Uber JAR,
```
Comment (lesv) [inline appengine-java11/README.md]: link to Uber definition . -- perhaps https://stackoverflow.com/questions/11947037/what-is-an-uber-jar

File: appengine-java11/appengine-simple-jetty-main/README.md [hunk 2]
```diff
@@ -1,5 +1,71 @@
...
+```
+runtime: java11
+instance_class: F2
+entrypoint: 'java -cp * com.example.appengine.demo.jettymain.Main helloworld.war'
+```
+
+## Running locally
+The `exec-maven-plugin` has been added so you can run your application locally.
+
+- Start the server with your `WAR` file as an argument:
```
Comment (lesv) [inline appengine-java11/appengine-simple-jetty-main/README.md]: While I can see why you chose to go this route, you might wish to explain why you aren't using the jetty-maven-plugin

File: appengine-java11/appengine-simple-jetty-main/pom.xml [hunk 3]
```diff
@@ -53,4 +55,25 @@
...
   </dependencies>
+  <!-- [END gae_java11_server_dependencies] -->
+
+  <build>
+    <plugins>
+      <!-- Exec Maven Plugin provides goals to help execute Main class locally -->
+      <!-- [START gae_java11_exec_plugin] -->
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>exec-maven-plugin</artifactId>
```
Comment (lesv) [inline appengine-java11/appengine-simple-jetty-main/pom.xml]: While it's probably ok, this plugin hasn't been updated in a long while and starts jetty inside the maven jvm.

File: appengine-java11/oauth2/pom.xml [hunk 4]
```diff
@@ -28,108 +28,109 @@
...
+        <artifactId>google-api-services-oauth2</artifactId>
+        <version>v2-rev131-1.23.0</version>
+        <scope>provided</scope>
+      </dependency>
+      <!-- [END gae_java11_oauth2_dependencies] -->
+
+      <dependency>
+        <groupId>javax.servlet</groupId>
+        <artifactId>javax.servlet-api</artifactId>
+        <version>4.0.1</version>
```
Comment (lesv) [inline appengine-java11/oauth2/pom.xml]: This should be 3.1.0 see https://www.eclipse.org/jetty/documentation/current/what-jetty-version.html

File: appengine-java11/README.md (hunk 1 above)
Comment (averikitsch) [reply appengine-java11/README.md]: done.

File: appengine-java11/appengine-simple-jetty-main/README.md (hunk 2 above)
Comment (averikitsch) [reply appengine-java11/appengine-simple-jetty-main/README.md]: done.

File: appengine-java11/oauth2/pom.xml (hunk 4 above)
Comment (averikitsch) [reply appengine-java11/oauth2/pom.xml]: done.

File: appengine-java11/oauth2/pom.xml [hunk 5]
```diff
@@ -28,108 +28,109 @@
...
+          <version>1</version>
+          <scope>provided</scope>
+      </dependency>
+
+      <!-- OAuth 2.0 packages -->
+      <!-- [START gae_java11_oauth2_dependencies] -->
+      <dependency>
+        <groupId>com.google.oauth-client</groupId>
+        <artifactId>google-oauth-client</artifactId>
+        <version>1.28.0</version>
```
Comment (lesv) [inline appengine-java11/oauth2/pom.xml]: 1.29.0 is current

File: appengine-java11/oauth2/pom.xml [hunk 6]
```diff
@@ -28,108 +28,109 @@
...
+        <groupId>com.google.oauth-client</groupId>
+        <artifactId>google-oauth-client-servlet</artifactId>
+        <version>1.28.0</version>
+        <scope>provided</scope>
+      </dependency>
+
+      <dependency>
+        <groupId>com.google.apis</groupId>
+        <artifactId>google-api-services-oauth2</artifactId>
+        <version>v2-rev131-1.23.0</version>
```
Comment (lesv) [inline appengine-java11/oauth2/pom.xml]: com.google.apis:google-api-services-oauth2:v2-rev20190313-1.28.0 . ??

File: appengine-java11/oauth2/pom.xml [hunk 7]
```diff
@@ -28,108 +28,109 @@
...
+        <groupId>com.google.oauth-client</groupId>
+        <artifactId>google-oauth-client</artifactId>
+        <version>1.28.0</version>
+        <scope>provided</scope>
+      </dependency>
+
+      <dependency>
+        <groupId>com.google.oauth-client</groupId>
+        <artifactId>google-oauth-client-servlet</artifactId>
+        <version>1.28.0</version>
```
Comment (lesv) [inline appengine-java11/oauth2/pom.xml]: 1.29.0

File: appengine-java11/oauth2/pom.xml [hunk 8]
```diff
@@ -28,108 +28,109 @@
...
+          <version>2.0.0</version>
+          <configuration>
+            <version>oauth2</version>
+          </configuration>
+        </plugin>
+
+        <plugin>
+          <groupId>org.eclipse.jetty</groupId>
+          <artifactId>jetty-maven-plugin</artifactId>
+          <version>9.4.2.v20170220</version>
```
Comment (lesv) [inline appengine-java11/oauth2/pom.xml]: Is this really the preferred?
org.eclipse.jetty:jetty-maven-plugin:9.4.18.v20190429

File: appengine-java11/appengine-simple-jetty-main/pom.xml [hunk 9]
```diff
@@ -53,4 +55,25 @@
       <version>9.4.14.v20181114</version>
```
Comment (lesv) [inline appengine-java11/appengine-simple-jetty-main/pom.xml]: org.eclipse.jetty:jetty-server:9.4.18.v20190429

Comment (averikitsch) [reply appengine-java11/appengine-simple-jetty-main/pom.xml]: done.

File: appengine-java11/oauth2/pom.xml (hunk 8 above)
Comment (averikitsch) [reply appengine-java11/oauth2/pom.xml]: done.

File: appengine-java11/oauth2/pom.xml (hunk 7 above)
Comment (averikitsch) [reply appengine-java11/oauth2/pom.xml]: done.

File: appengine-java11/oauth2/pom.xml (hunk 6 above)
Comment (averikitsch) [reply appengine-java11/oauth2/pom.xml]: done. Next time I will remember to check the dependencies. Thanks!

**Pull request merged**